src/async_impl/response.rs

```diff
@@ -95,6 +97,25 @@ impl Response {
     }
 }
 
+impl Read for Response {
```

Comment:
I'd rather not impl `Read` for the async `Response`, I think. Maybe it does make sense, but probably a separate issue.